src/ecs/systems/ManageClientEvents.hpp

```diff
@@ -60,42 +71,11 @@ namespace ecs::systems
-                auto const &controllables = world.registry.getComponents<component::Controllable>();
-                auto const &positions = world.registry.getComponents<component::Position>();
-                auto &weapons = world.registry.getComponents<component::Weapon>();
-                auto const &factions = world.registry.getComponents<component::Faction>();
-
-                for (size_t i = 0; i < positions.size() && i < controllables.size(); ++i) {
-                    auto const &pos = positions[i];
-                    auto const &con = controllables[i];
-                    auto &weapon = weapons[i];
-                    if (pos && con && weapon) {
-                        auto elapsed = constant::chrono::now().time_since_epoch().count() - weapon.value().lastShoot;
-                        if (weapon.value().hasSuper && elapsed > weapon.value().superLoadingTime) {
-                            weapon.value().lastShoot = constant::chrono::now().time_since_epoch().count();
-                            // spawn super bullet
-                        } else if (elapsed > weapon.value().shootDelay) {
-                            weapon.value().lastShoot = constant::chrono::now().time_since_epoch().count();
-                            ecs::Entity bullet = world.registry.spawn_entity();
-                            world.registry.addComponent<ecs::component::Direction>(bullet, {1, 0});
-                            world.registry.addComponent<ecs::component::Position>(
-                                bullet, {pos.value().x, pos.value().y});
-                            world.registry.addComponent<ecs::component::Size>(bullet, {10, 10});
-                            world.registry.addComponent<ecs::component::Velocity>(bullet, {weapon.value().projSpeed, 0});
-                            world.registry.addComponent<ecs::component::Projectile>(bullet, {weapon.value().damage});
-                            ecs::component::Faction::Factions fac = ecs::component::Faction::Factions::None;
-                            if (i < factions.size() && factions[i])
-                                fac = factions[i].value().faction;
-                            world.registry.addComponent<ecs::component::Faction>(bullet, {fac});
-#ifdef CLIENT_COMPILATION_MODE
-                            std::filesystem::path playerPath =
-                                ecs::crossPlatformPath("src", "demo", "assets", "textures", "players.gif");
-                            world.registry.addComponent<ecs::component::Drawable>(
-                                bullet, {playerPath, {5, 5, 1, 1}}); // To be removed with the server/client split
-#endif
-                        }
-                    }
-                }
+                std::cout << "SHOOOOOOOOOOOOOOOOOT" << std::endl;
+                network::Message msg;
+                msg.fill(0);
+                msg.at(0) = constant::getPacketTypeKey(ecs::constant::PacketType::PLAYER_SHOT);
```

Comment:
I'm not sure that you want to put an integer into an array of char

Reply:
I'm sure I want